### PR TITLE
Buildsystem improvements 240116

### PR DESCRIPTION
**Pull request by uyjulian**
The major change is that this fixes the race condition when installing samples with multiple jobs

### CODE REVIEW

File: .github/workflows/compilation.yml
Comment (fjtrujy) [inline .github/workflows/compilation.yml:1]: Shouldn't you also to put the `$(getconf _NPROCESSORS_ONLN)` instead `$(getconf _NPROCESSORS_ONLN)`

Comment (fjtrujy) [reply .github/workflows/compilation.yml:1]: And the same in the `dockerfile`

Comment (uyjulian) [reply .github/workflows/compilation.yml:1]: They are already POSIX shell expansion, so I don't need to do anything with that.

Comment (fjtrujy) [reply .github/workflows/compilation.yml:1]: Sorry I wrote it wrong previously, I wanted to say that we still have hardcoded `-j 1` in the workflow and docker file

Comment (uyjulian) [reply .github/workflows/compilation.yml:1]: An. Yeah I could do that

**Pull request merged**